src/SpacetimeDBClient.cs

```diff
@@ -523,19 +572,16 @@ private void OnMessageProcessCompleteUpdate(ReducerEvent? dbEvent, List<DbOp> db
                     {
                         case { insert: { value: var newValue }, delete: { value: var oldValue } }:
                             {
-                                // If we matched an update, these values must have primary keys.
-                                var newValue_ = (IDatabaseTableWithPrimaryKey)newValue;
-                                var oldValue_ = (IDatabaseTableWithPrimaryKey)oldValue;
-                                oldValue_.OnUpdateEvent(newValue_, dbEvent);
+                                dbOp.table.Handle.InvokeUpdate(eventContext, oldValue, newValue);
```

Comment:
It's not a big deal, but FWIW a separate interface and the cast were there to ensure that we don't accidentally emit update events on tables that don't have a primary key, since it's not valid to do so.

In principle, this shouldn't happen if we reached here, but 🤷‍♂️

Reply:
Ah yeah that's tricky now because the type can represent both a table with a PK and one without one, or two tables using different columns as their PKs.

Besides, the only way for updates to be collected is for their PKs to have been collected through the handle previously, which performs the check we want.